src/ReturnTypes/Helpers/AppExtension.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NunoMaduro\Larastan\ReturnTypes\Helpers;
+
+use Illuminate\Foundation\Application;
+use NunoMaduro\Larastan\Concerns\HasContainer;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Type\DynamicFunctionReturnTypeExtension;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\MixedType;
+use PHPStan\Type\NullType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use Throwable;
+
+class AppExtension implements DynamicFunctionReturnTypeExtension
+{
+    use HasContainer;
+
+    public function isFunctionSupported(FunctionReflection $functionReflection): bool
+    {
+        return $functionReflection->getName() === 'app' || $functionReflection->getName() === 'resolve';
+    }
+
+    public function getTypeFromFunctionCall(
+        FunctionReflection $functionReflection,
+        FuncCall $functionCall,
+        Scope $scope
+    ): Type {
+        if (count($functionCall->args) === 0) {
+            return new ObjectType(Application::class);
+        }
+
+        /** @var Expr $expr */
+        $expr = $functionCall->args[0]->value;
+
+        if ($expr instanceof String_) {
+            try {
+                $resolved = $this->resolve($expr->value);
+
+                if (is_null($resolved)) {
+                    return new NullType();
+                }
+
+                return new ObjectType(get_class($resolved));
+            } catch (Throwable $exception) {
+                return new ErrorType();
+            }
+        }
+
+        if ($expr instanceof ClassConstFetch && $expr->class instanceof FullyQualified) {
+            return new ObjectType($expr->class->toString());
+        }
+
+        return new MixedType();
```

Comment:
We should always try to avoid the Mixed type.

Reply:
Is it safe to return `ErrorType` here also? Because if we are here this far, it means nothing is resolved from the container. In normal usage, an exception would be thrown.

Reply:
Exactly. That is the goal of static analysis. Prevent runtime errors 👍

Reply:
This needs to be removed.

Reply:
Actually it might be ok here. An `ErrorType` or `NeverType` would be the same here. Because if we are here this far it means nothing is resolved from the container. 

And there is no alternative to do here.

Reply:
`ErrorType` can make more sense semantically.

So if you can change this here and rebase the branch, I think we are good to go with this :+1: